apps/docs/pages/guides/platform/custom-domains.mdx

```diff
@@ -18,139 +18,213 @@ Custom domains allow you to present a branded experience to your users. Custom d
-  ```
+1. Go through all of your Supabase OAuth providers:
+   - In the provider's developer console (not in the Supabase dashboard!), find the OAuth application and add the custom domain Supabase Auth callback URL **in addition to the Supabase project URL.** Example:
+     - `https://abcdefghijklmnopqrst.supabase.co/auth/v1/callback` **and**
+     - `https://api.example.com/auth/v1/callback`
+   - [Sign in with Twitter](/docs/guides/auth/social-login/auth-twitter) uses cookies bound to the project's domain. In this case make sure your frontend code uses the custom domain instead of the default project's domain.
+2. Go through all of your SAML identity providers:
+   - You will need to reach out via email to all of your existing identity providers and ask them to update the metadata for the SAML application (your project). Use `https://api.example.com/auth/v1/...` instead of `https://abcdefghijklmnopqrst.supabase.co/auth/v1/callback`.
+   - Once these changes are made, SAML Single Sign-On will likely stop working until the domain is activated. Plan for this ahead of time.
 
-- Any OAuth providers (e.g., google, github)
+### Activate your domain
 
-Additionally, update the DNS configuration for `api.example.com` to once more use a CNAME record that resolves to `foobarbaz.supabase.co`.
-Finally, you can use the `activate` subcommand to reconfigure your project:
+Once you've done the necessary preparations to activate the new domain for your project, you can activate it using the [`domains activate`](/docs/reference/cli/supabase-domains-activate) CLI command.
 
 ```bash
-supabase domains activate --project-ref foobarbaz --experimental
+supabase domains activate --project-ref abcdefghijklmnopqrst
 ```
 
-## Remove a Custom Domain
+When this step completes, Supabase will serve the requests from your new domain. The Supabase project domain **continues to work** and serve requests so you do not need to rush to change client code URLs.
 
-If you have a custom domain (`api.example.com`) set up for your Supabase project (ref `foobarbaz`, with assigned endpoints at `foobarbaz.supabase.co`), and would like to go back to using the Supabase-provisioned endpoints (`foobarbaz.supabase.co`), you can use the `delete` subcommand:
+If you wish to use the new domain in client code, you can set it up like so:
 
-```bash
-supabase domains delete --project-ref foobarbaz --experimental
+```js
+import { createClient } from '@supabase/supabase-js'
+
+// Use a custom domain as the supabase URL
+const supabase = createClient('https://api.example.com', 'public-anon-key')
 ```
 
-As with the final activation stage of the process for setting up a custom domain, you'll need to update any references in your client code and OAuth providers from the custom domain to the Supabase-provisioned endpoints.
+When using [Sign in with Twitter](/docs/guides/auth/social-login/auth-twitter) make sure your frontend code is using the custom domain only!
```

Comment:
```suggestion
```

it's mentioned above so I would lean towards removing this, or perhaps throwing it into an admonition, it seems a bit out of place here